src/weight_windows.cpp

```diff
@@ -0,0 +1,284 @@
+#include "openmc/search.h"
+#include "openmc/weight_windows.h"
+#include "openmc/xml_interface.h"
+
+namespace openmc {
+
+// read the weight window file
+void read_weight_window_xml() {
+  using namespace weight_window;
+  using namespace settings;
+  using namespace pugi;
+
+  // Check if weight_window.xml exists                                                         
+  std::string filename = path_input + "weight_windows.xml";
+  
+  // Parse settings.xml file                                                              
+  xml_document doc;                                                                       
+  if (file_exists(filename)) {
+    auto result = doc.load_file(filename.c_str());                                          
+    if (!result) {                                                                          
+      fatal_error("Error processing weight_windows.xml file.");                                   
+    }
+    // Get root element                                                                     
+    xml_node root = doc.document_element();     
+    std::cout << " Reading weight window XML file ..." << std::endl;
+    // try and read a weight window
+    settings::ww_settings = std::make_shared<weight_window::WeightWindow>(root);
+  } else {
+    settings::ww_settings = std::make_shared<weight_window::WeightWindow>();
+  }                                                                       
+}
+
+//==============================================================================
+// WeightWindowMesh implementation
+//==============================================================================
+
+using namespace weight_window;
+
+WeightWindow::WeightWindow() 
+{
+  // default parameters
+  n_ww = false;                  // flag for neutron use weight window
+  p_ww = false;                  // flag for photon use weight window
+  user_defined_biasing = false;  // flag for weight biasing in energy
+
+  // upper weight window = upper_ratio * lower weight window
+  weight_params[ParticleType::neutron].upper_ratio = 5.; 
+  // survival weight = survival_ratio * lower weight window  
+  weight_params[ParticleType::neutron].survival_ratio = 3.; 
+  // max number of split particles
+  weight_params[ParticleType::neutron].max_split = 5; 
+  // multiplier for weight window lower bounds
+  weight_params[ParticleType::neutron].multiplier = 1.0; 
+  
+  // upper weight window = upper_ratio * lower weight window
+  weight_params[ParticleType::photon].upper_ratio = 5.; 
+  // survival weight = survival_ratio * lower weight window  
+  weight_params[ParticleType::photon].survival_ratio = 3.; 
+  // max number of split particles
+  weight_params[ParticleType::photon].max_split = 5; 
+  // multiplier for weight window lower bounds
+  weight_params[ParticleType::photon].multiplier = 1.0; 
+
+}
+
+WeightWindow::WeightWindow(pugi::xml_node node)
+{
+  using namespace pugi;
+
+  WeightWindow(); // call the default constructor
```

Comment:
Don't do this here. It's not actually doing what you think -- this is creating a `WeightWindow` object here that never gets used or assigned to any variable. What you want is a "delegating constructor", please look it up.